internal/controller/driver_controller.go

```diff
@@ -981,7 +982,41 @@ func (r *driverReconcile) reconcileNodePluginDeamonSetForCsiAddons() error {
 
 	log := r.log.WithValues("csiAddonsDaemonSetName", daemonSet.Name)
 
-	if !ptr.Deref(r.driver.Spec.DeployCsiAddons, false) {
+	// FIXME: Spec.DeployCsiAddons defaults to true, causes CSI-Addons for CephFS enabled by default
+	withCsiAddonsDaemonSet := false // ptr.Deref(r.driver.Spec.DeployCsiAddons, false)
+	withCsiAddonsVolumeCondition := false
+
+	// FIXME: this workaround allows a user to disable the DaemonSet for RBD
```

Comment:
@nixpanic This is not a workaround i think its the actual behaviour

Reply:
Yeah, well, using `ptr.Deref(r.driver.Spec.DeployCsiAddons, false)` should be done above:

```go
withCsiAddonsDaemonSet := ptr.Deref(r.driver.Spec.DeployCsiAddons, false)
```

and then this if statement can be removed. But that requires `ptr.Deref(r.driver.Spec.DeployCsiAddons, false)` to return `false` if it isn't set.